common/component/kafka/metadata.go

```diff
@@ -129,25 +129,25 @@ func (k *Kafka) upgradeMetadata(metadata map[string]string) (map[string]string,
 				// If legacy authRequired was used, either SASL username or mtls is the method.
 				if saslPassPres && saslPassVal != "" {
 					// User has specified saslPassword, so intend for password auth.
-					metadata[authType] = passwordAuthType
+					meta[authType] = passwordAuthType
 				} else {
-					metadata[authType] = mtlsAuthType
+					meta[authType] = mtlsAuthType
 				}
 			} else {
-				metadata[authType] = noAuthType
+				meta[authType] = noAuthType
```

Comment:
Don't update `meta`, because it can cause issues if the key had different casing.

Pass the auth type in a separate variable.

Reply:
this is all preexisting code... but i can update, sure

Reply:
Clarified in discord